src/useSpeech/index.ts

```diff
@@ -0,0 +1,117 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { useIsMounted } from '../useIsMounted';
+
+type SpeechOptions = {
+	voice?: SpeechSynthesisVoice;
+	rate: number;
+	pitch: number;
+	volume: number;
+	lang?: string;
+};
+
+export type ISpeechOptions = Partial<SpeechOptions>;
+
+export type VoiceInfo = Pick<SpeechSynthesisVoice, 'lang' | 'name'>;
+
+type SpeechPlayStatus = 'init' | 'play' | 'pause' | 'end' | 'error';
+
+export type ISpeechState = SpeechOptions & {
+	isPlaying: boolean;
+	status: SpeechPlayStatus;
+	voiceInfo: VoiceInfo;
+	errorMessage?: string;
+};
+
+/**
+ * Hook for playing text using the SpeechSynthesis API.
+ *
+ * @param {string} text - Text to play.
+ * @param {ISpeechOptions} options - Options for playing the text.
+ * @param {SpeechSynthesisVoice} [options.voice] - Voice to use for playing the text.
+ * @param {number} [options.rate] - Rate at which to play the text.
+ * @param {number} [options.pitch] - Pitch at which to play the text.
+ * @param {number} [options.volume] - Volume at which to play the text.
+ * @param {string} [options.lang] - Language of the text.
+ * @returns {ISpeechState} State object representing the current state of speech synthesis.
+ */
+
+export const useSpeech = (text: string, options: ISpeechOptions): ISpeechState => {
+	const isMounted = useIsMounted();
+	const mounted = isMounted();
+	const [state, setState] = useState<ISpeechState>(() => {
+		const { lang = 'default', name = '' } = options.voice ?? {};
+		return {
+			isPlaying: false,
+			status: 'init',
+			voiceInfo: {
+				lang,
+				name,
+			},
+			rate: options.rate ?? 1,
+			pitch: options.pitch ?? 1,
+			volume: options.volume ?? 1,
+		};
+	});
+
+	const handlePlay = useCallback(() => {
```

Comment:
no need in separate `useCallbacks` - use singular `useMemo` instead

Reply:
@xobotyi Not understood what are you trying to say here.  Could you please elaborate here?

Added `useCallback` just because it doesn't change it's reference since it is been added in `useEffect`